chat_shell/tests/test_kb_prompt_deduplication.py

```
@@ -0,0 +1,259 @@
+    async def test_empty_kb_ids_with_skip_skips_historical_meta(self):
+        """When no KB IDs and skip=True, should skip historical KB meta prompt."""
+        from chat_shell.tools.knowledge_factory import prepare_knowledge_base_tools
+
+        base_prompt = "This is the base system prompt."
+
+        with patch(
+            "chat_shell.tools.knowledge_factory._build_historical_kb_meta_prompt"
+        ) as mock_meta:
+            mock_meta.return_value = "\nHistorical KB meta"
+
+            tools, enhanced_prompt = await prepare_knowledge_base_tools(
```

Ruff (0.14.13)

Unpacked variable `tools` is never used

Prefix it with an underscore or any other dummy variable pattern

(RUF059)